python/strings_udf/cpp/src/strings/udf/shim.cu

```diff
@@ -270,3 +271,57 @@ extern "C" __device__ int rstrip(int* nb_retval,
 
   return 0;
 }
+
+extern "C" __device__ int upper(int* nb_retval,
+                                void* udf_str,
+                                void* const* st,
+                                std::int64_t flags_table,
+                                std::int64_t cases_table,
+                                std::int64_t special_table)
```

Comment:
These are pointers being represented by integers I think. So you want `std::uintptr_t` not `int64_t` (although it is only optional in C++11 and later, I think it is provided on all sane platforms).

This comment applies throughout (and indeed in existing code where you accept pointers as integers and then call `reinterpret_cast<some_ptr*>(some_integer)`)